src/formatDate.js

```diff
@@ -8,7 +8,28 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+
+  const dateParts = date.split(fromSeparator);
```

Comment:
It's good practice to handle potential issues that could arise if the date string does not contain the expected number of parts after splitting. You can add a check to ensure that the `dateParts` array has the same length as the `fromFormat` array minus the separator.